dashboard/app/models/plc/enrollment_unit_assignment.rb

```diff
@@ -36,6 +36,9 @@ def enroll_user_in_unit_with_learning_modules(learning_modules)
     transaction do
       plc_module_assignments.destroy_all
 
+      # Make sure the required learning modules are included here
+      learning_modules |=  self.plc_course_unit.plc_learning_modules.where(required: true)
```

Comment:
Ah, TIL.  I've seen `array_a | array_b` before but this is the first time I've seen `|=`.